src/opm/input/eclipse/Parser/ParserKeyword.cpp

```diff
@@ -63,35 +63,34 @@ KeywordSize::KeywordSize(std::size_t in_min_size, const std::string& in_keyword,
 
 KeywordSize::KeywordSize()
     : KeywordSize(SLASH_TERMINATED)
-{
-}
+{}
 
 KeywordSize::KeywordSize(ParserKeywordSizeEnum size_type)
     : m_size_type(size_type)
 {
-    if (size_type == SLASH_TERMINATED)
-        return;
-
-    if (size_type == UNKNOWN)
-        return;
-
-    if (size_type == DOUBLE_SLASH_TERMINATED)
+    if ((size_type == ParserKeywordSizeEnum::SLASH_TERMINATED) ||
+        (size_type == ParserKeywordSizeEnum::UNKNOWN) ||
+        (size_type == ParserKeywordSizeEnum::DOUBLE_SLASH_TERMINATED) ||
+        (size_type == ParserKeywordSizeEnum::SPECIAL_CASE_ROCK))
+    {
         return;
+    }
 
-    throw std::logic_error("This constructor only allows size type UNKNOWN and SLASH_TERMINATED");
+    throw std::logic_error {
+        "This constructor only allows size types "
+        "UNKNOWN, SLASH_TERMINATED, and SPECIAL_CASE_ROCK"
```

Comment:
... and DOUBLE_SLASH_TERMINATED

Reply:
> ... and DOUBLE_SLASH_TERMINATED

Sure, but we don't actually have any of those.

Reply:
> Wouldn't it be nice to reflect that in the error message?

I've pushed an update to this effect.